### ISSUE TITLE
Example for showing/hiding columns?

### ISSUE THREAD

**Issue by TomLinthwaite** [vgrem/Office365-REST-Python-Client#394]
Hi, is there any chance you can provide an example of how to show fields after adding them to a list? I found this [blog post](https://www.sharepointdiary.com/2016/01/sharepoint-online-hide-column-from-content-type-using-powershell.html) that seems to do what I want but I'm struggling to do it with this API.

Apologies, new to REST and Python, I have tried digging around the source as well and can find the objects I just cant seem to put it together.

Thank you

**Comment by vgrem** [comment]
Greetings,

at least the following options are available to control visibility (namely _hide_ in below examples) fields in `List`:




### Example 1

As you've suggested, by setting `Field.hide` property which [per documentation](https://docs.microsoft.com/en-us/previous-versions/office/sharepoint-csom/ee540624(v=office.15)):

> Gets or sets a value that specifies whether the field is hidden in list views and list forms.  

```python 
ctx = ClientContext(site_url).with_credentials(client_credentials)

target_list = ctx.web.lists.get_by_title("Tasks")
target_field = target_list.fields.get_by_internal_name_or_title("AssignedTo")

target_field.hidden = True
target_field.update().execute_query()
```


### Example 2

or a per list form (namely `Display`, `Edit` and `New` form types) 

Refer [`Field.SetShowInDisplayForm` method](https://docs.microsoft.com/en-us/previous-versions/office/sharepoint-csom/jj862540(v=office.15)) documentation for a more details 


```python
ctx = ClientContext(site_url).with_credentials(client_credentials)

target_list = ctx.web.lists.get_by_title("Tasks")
target_field = target_list.fields.get_by_internal_name_or_title("AssignedTo").get().execute_query()

target_field.set_show_in_new_form(True)  #hide from New form
target_field.set_show_in_edit_form(True) #hide from Edit form
target_field.set_show_in_display_form(True) #hide from Display form
ctx.execute_query()

```



If is not

**Comment by TomLinthwaite** [comment]
Thank you for replying so quickly, but I must have asked the wrong question because I didn't understand what I needed. i tried the above and I still failed to see my columns displayed when I visited the web page itself. I had another dig around and it seems like [this solution](https://powerusers.microsoft.com/t5/Building-Flows/Using-REST-API-to-create-a-field-in-a-Sharepoint-list-all/td-p/1047712) might work? I tried to figure it out using the API and inspecting a list in debug mode but got as far as finding the views but no 'viewFields'/'addViewFields' function or property. Thanks again.
